### PR TITLE
BT GATT macro for multi-instance service definitions

### PR DESCRIPTION
**Pull request by kapi-no**
This PR completes a side task that was brought up in this review comment:
https://github.com/zephyrproject-rtos/zephyr/pull/27088#discussion_r480858262

### CODE REVIEW

**Review by Vudentz — APPROVED** [review]
LGTM

**Comment by kapi-no** [comment]
Related with #27088

File: include/bluetooth/gatt.h [hunk 1]
```diff
@@ -452,6 +452,58 @@ ssize_t bt_gatt_attr_read_service(struct bt_conn *conn,
...
+
+/** @def _BT_GATT_SERVICE_ARRAY_ITEM
+ *  @brief Single item listed in the service structure array.
+ *
+ *  Helper macro to declare single item listed in the service structure array.
+ *  It is used in the @ref BT_GATT_SERVICE_INSTANCE_DEFINE macro.
+ *
+ *  @param _n Item number.
+ */
+#define _BT_GATT_SERVICE_ARRAY_ITEM(_n, _) BT_GATT_SERVICE(attrs_##_n),
```
Comment (Vudentz) [inline include/bluetooth/gatt.h:479]: I guess these one don't really need to be exposed in the documentation, or are you planning on using them elsewhere?

Comment (kapi-no) [reply include/bluetooth/gatt.h:479]: No, this is a internal macro that should only be used by BT_GATT_SERVICE_INSTANCE_DEFINE. I can remove the documentation completely

**Review by joerchan — APPROVED** [review]

**Pull request merged**